Update readme (#1)
mestcihazal/mcp3004-8-go: README.md

> # `mcp3004-8-go` modular resource

Author: ```suggestion
# `mcp3004-8` modular resource
```

This module provides analog-to-digital conversion capabilities for MCP3004 and MCP3008 SPI ADCs. Written in Go, board agnostic. Tested on Raspberry Pi and Jetson Orin.

The values you get from the readings will be proportional to the voltage and you will need to interpret it depending on which sensor you hook up to the MCP300x.

You can add as many analog sensors as your MCP300x allows and get readings from them concurrently (this depends on how many channels it has, so for MCP3004 that is four channels, and for MCP3008 that is eight).

## Build and run

> To use this module, follow the instructions to [add a module from the Viam Registry](https://docs.viam.com/registry/configure/#add-a-modular-resource-from-the-viam-registry) and select the `hazalmestci:sensor:mcp3004-8-go` model from the [`viam-mcp3004-8-go` module](https://app.viam.com/module/hazalmestci/mcp3004-8).

Author: ```suggestion
To use this module, follow the instructions to [add a module from the Viam Registry](https://docs.viam.com/registry/configure/#add-a-modular-resource-from-the-viam-registry) and select the `hazalmestci:sensor:mcp3004-8-go` model from the [`mcp3004-8` module](https://app.viam.com/module/hazalmestci/mcp3004-8).
```

> ## Configure your `mcp3004-8-go`

Author: ```suggestion
## Configure your `mcp3004-8`
```

> [!NOTE]

> > Before configuring your `mcp3004-8-go`, you must [create a machine](https://docs.viam.com/manage/fleet/machines/#add-a-new-machine).

Author: ```suggestion
> Before configuring your `mcp3004-8`, you must [create a machine](https://docs.viam.com/manage/fleet/machines/#add-a-new-machine).
```

    "humidity": 2
  }
}
```

Save your config.

### Attributes

> The following attributes are available for a `viam-mcp3004-8-go` sebsor:

Author: ```suggestion
The following attributes are available for a `mcp3004-8` sensor:
```

| Name    | Type   | Inclusion    | Description |
| ------- | ------ | ------------ | ----------- |
| `chip_select` | string | **Required** | The `chip_select`` pin you are using. |
| `spi_bus` | string | **Required** | the `spi_bus` you are using. |
| `pins` | string | **Required** | The pins you are using for moisture, temperature, and humidity. |

### Example configuration

```json
{
    "name": "my-mcp3004",
    "model": "hazalmestci:sensor:mcp3004-8-go",
    "type": "sensor",
    "namespace": "rdk",
    "attributes": {
      "chip_select": "0",
      "spi_bus": "0", 
      "pins": {
        "moisture": 0,
        "temperature": 1,
        "humidity": 2
      }
    },
    "depends_on": []
}
```

## Local Development

> To use the `viam-mcp3004-8-go` module with a local install, clone this repository to your machine’s computer, navigate to the `module` directory, and run:

Author: ```suggestion
To use the `mcp3004-8` module with a local install, clone this repository to your machine’s computer, navigate to the `module` directory, and run:
```
